the-collab-lab/tcl-37-smart-shopping-list
# Join Existing List

## Description
Tokens can be shared with other users to allow them to co-manage a given list.
Joining an existing list consists of the following:
* User enters an existing token
* Save the token to localStorage
* Show the user the existing list

From the Home page, we render the JoinList component, passing down the setToken function, as a prop. We are passing to later use it in our handleSubmit to share the token state in App.js to make it universal for the whole app.

Join List is a form that allows the user to enter and submit a token. This token is used to update the state of the token at the App level, through the setToken function, as we pass it the user input. 

When the state of the token changes, the user is directed to the List component where they can add items to their existing list.

The shareable list token is now available from the list view, as well as a link to navigate back to Home.



## Related Issue
Closes #5 


## Acceptance Criteria

- [x] When a user is viewing a list, there should be a mechanism for them to access the token so they can share it with someone else

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |



## Updates

### Before
<img width="871" alt="Screen Shot 2022-01-28 at 1 53 00 PM" src="https://user-images.githubusercontent.com/67841237/151604927-38d217e4-730e-4d82-b4cc-6cd8dbbc54a7.png">




### After

https://user-images.githubusercontent.com/67841237/151612875-1c43af0a-3e9d-4a11-9630-2ca2415bb97b.mp4




## Testing Steps / QA Criteria

1 . npm install 
2.  npm start
3.  From the Home page, you can enter a token. One that already exists is "vary adore suite", enter the token in the input box and submit. You should expect to be directed to the ListView for that collection.
4.  Go ahead and navigate back home. Enter any characters (that do not exactly correspond to a collection) in the Share token box. You should expect to see an error message pop up on the screen.
5. Try again to enter a token. You can delete all the characters without throwing an error.
6.  Try another existing token "alba terre tussle" and submit. You should expect to see the list for that collection. On the ListView you will see the shareable token a user can provide to enable someone else to access the collection.
